com.unity.ml-agents/Runtime/Communicator/GrpcExtensions.cs

```diff
@@ -310,7 +333,29 @@ public static UnityRLCapabilitiesProto ToProto(this UnityRLCapabilities rlCaps)
             {
                 BaseRLCapabilities = rlCaps.BaseRLCapabilities,
                 ConcatenatedPngObservations = rlCaps.ConcatenatedPngObservations,
+                CompressedChannelMapping = rlCaps.CompressedChannelMapping,
             };
         }
+
+        private static bool IsTrivialMapping(ISensor sensor)
+        {
+            var compressibleSensor = sensor as ISparseChannelSensor;
+            if (compressibleSensor is null)
+            {
+                return true;
+            }
+            var mapping = compressibleSensor.GetCompressedChannelMapping();
+            if (mapping == null)
+            {
+                return true;
+            }
+            var zeroMapping = new int[] { 0, 0, 0 };
```

Comment:
nit: can you just check these with a loop? This will allocate memory each step, which will impact performance.

I also think you can only treat {0,0,0} as trivial if numChannels==1, right?

Reply:
{0,0,0} can only have numChannels==1 no?

Reply:
Hmm... I could write a ISparseChannelSensor that returns {0,0,0} for it's mapping but has shape[2] = 2. Not sure what would happen on the trainer side - maybe all 0s? It's probably another error condition that we need to check for somewhere.

Related, we should also have a check that the mapping values are less than the number of channels. For example, {0, 1, 99} with numChannels == 3 should be an error (and will probably raise an exception on python).

Reply:
On python side, if there's a mapping sent it always checks if the mapping is the same length as the input observation channels, and raises an exception if it's not. So in your case it will raise exception. Since mapping is designed for non-trivial case I don't think it's a good idea to allow partial mapping here.

Thanks for remind, will add the check for mapping values less than the number of channels.